client/chain_registry/chain_info.go

```diff
@@ -69,10 +69,17 @@ func (c ChainInfo) GetAllRPCEndpoints() (out []string, err error) {
 		var port string
 		if u.Scheme == "https" {
 			port = "443"
+		} else if u.Scheme == "http" && u.Port() == "" {
+			port = "80"
 		} else {
 			port = u.Port()
 		}
-		out = append(out, fmt.Sprintf("%s://%s:%s", u.Scheme, u.Hostname(), port))
+
+		if u.Path == "" {
+			out = append(out, fmt.Sprintf("%s://%s:%s", u.Scheme, u.Hostname(), port))
+		} else {
+			out = append(out, fmt.Sprintf("%s://%s:%s%s", u.Scheme, u.Hostname(), port, u.Path))
+		}
```

Comment:
I believe you can simplify this conditional to:
```suggestion
			out = append(out, fmt.Sprintf("%s://%s:%s%s", u.Scheme, u.Hostname(), port, u.Path))
```

Reply:
Very true. Fixed (reverted) this change.